circuit_maintenance_parser/parser.py

```diff
@@ -138,13 +138,20 @@ def parse_ical(gcal: Calendar) -> List[Dict]:
                         )
                         for object in component.get("X-MAINTNOTE-OBJECT-ID")
                     ]
-                else:
+                elif circuits is not None:
                     data["circuits"] = [
                         CircuitImpact(
                             circuit_id=circuits,
                             impact=Impact(component.get("X-MAINTNOTE-IMPACT")),
                         )
                     ]
+                elif data["status"] == "CANCELLED":
+                    # We shouldn't have to do this, as it does not
```

Comment:
trying to understand...
EUNetworks is sending a notification with a maintenance ID about canceling it, but it's not providing any information about the circuit IDs, which were referenced in the initial notification message. isn't it?

Reply:
Yes, you understand properly.  The actual maintenance notification draft document however says that most information should be copied from the previous notification when sending a cancelled status - https://datatracker.ietf.org/doc/html/draft-gunter-calext-maintenance-notifications-00#section-3.3 - so hopefully this is rare behavior for at least the people following that draft (I've only seen it from EUNetworks but that of course doesn't mean nobody else is doing it).

I'll submit an update to this PR that does this as you describe.